cmd/main.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"flag"
+	"log"
+	"os"
+	"time"
+
+	pkg "github.com/codescalersinternships/Datetime-client-RawanMostafa/pkg"
+)
+
+const defaltBaseUrl = "http://localhost"
+const defaultEndpoint = "/datetime"
+const defaultPort = "8083"
+
+func getFlags() (string, string, string) {
+	var port string
+	flag.StringVar(&port, "port", "", "Specifies the port")
+
+	var baseUrl string
+	flag.StringVar(&baseUrl, "baseUrl", "", "Specifies the base url")
+
+	var endpoint string
+	flag.StringVar(&endpoint, "endpoint", "", "Specifies the endpoint")
+
+	flag.Parse()
+	return baseUrl, endpoint, port
+}
+
+func decideConfigs() (string, string, string) {
+
+	baseUrl, endpoint, port := getFlags()
+
+	if baseUrl == "" {
+		envBaseUrl, found := os.LookupEnv("DATETIME_BASEURL")
+
+		if found {
+			baseUrl = envBaseUrl
+		} else {
+			baseUrl = defaltBaseUrl
+		}
```

Comment:
we can skip the extra else like this
```suggestion
		var found bool
		baseUrl, found = os.LookupEnv("DATETIME_BASEURL")
		if !found {
			baseUrl = defaltBaseUrl
		}

```

Reply:
same for all